kernel/aster-nix/src/vm/vmo/static_cap.rs

```diff
@@ -6,74 +6,17 @@ use aster_rights::{Dup, Rights, TRightSet, TRights, Write};
 use aster_rights_proc::require;
 use ostd::mm::{Frame, VmIo};
 
-use super::{
-    options::{VmoCowChild, VmoSliceChild},
-    CommitFlags, Vmo, VmoChildOptions, VmoRightsOp,
-};
+use super::{CommitFlags, Vmo, VmoRightsOp};
 use crate::prelude::*;
 
 impl<R: TRights> Vmo<TRightSet<R>> {
-    /// Creates a new slice VMO through a set of VMO child options.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// let parent = VmoOptions::new(PAGE_SIZE).alloc().unwrap();
-    /// let child_size = parent.size();
-    /// let child = parent.new_slice_child(0..child_size).alloc().unwrap();
-    /// assert!(child.size() == child_size);
-    /// ```
-    ///
-    /// For more details on the available options, see `VmoChildOptions`.
-    ///
-    /// # Access rights
-    ///
-    /// This method requires the Dup right.
-    ///
-    /// The new VMO child will be of the same capability flavor as the parent;
-    /// so are the access rights.
-    #[require(R > Dup)]
-    pub fn new_slice_child(
```

Comment:
The commit "Decouple Vmo from VmMapping" should have been broken into atomic ones. It includes at least four separate atomic commits:

* Do some clean-up for `vdso.rs`
* Remove the concept of child VMOs
* Create anonymous memory mappings without VMOs
* Support shared memory mappings

Putting the non-atomic problem aside, the commit message "Decouple Vmo from VmMapping" is very confusing. None of the above commits can be described as "Decouple Vmo from VmMapping". `Vmo` is still being used by `VmMapping`, just to a lesser extent. As long as `Vmo` is known and used by `VmMapping`, one cannot say that `Vmo` is decoupled from `VmMapping`.

It is important to make _commits_ atomic because each commit is supposed to make changes for one specific purpose. When I read this commit, I had no idea about the motivation of the changes. The only (and misleading) hint is to _decouple_. I have to "decipher" the motivation of the changes for different purposes. If this commit was broken into four atomic ones, it would have been much easier for my to grasp its intention.